app/scripts/modules/core/src/presentation/hooks/usePollingData.hook.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+import { IPromise } from 'angular';
+import { SchedulerFactory } from 'core/scheduler';
+
+import { useLatestCallback } from './useLatestCallback.hook';
+import { IUseLatestPromiseResult } from './useLatestPromise.hook';
+import { useData } from './useData.hook';
+
+/**
+ * A react hook which invokes a promise factory callback whenever any of its dependencies
+ * changes *and* on a specific polling schedule.
+ *
+ * Returns the default result until the first promise resolves.
+ *
+ * The promise factory is not called if any of the deps are null or undefined.
+ *
+ * This can be useful when fetching data based on a users keyboard input, for example.
+ * This behavior is similar to RxJS switchMap.
+ *
+ * @param callback the callback to be invoked whenever dependencies change and on a polling schedule
+ * @param defaultResult the default result returned before the first promise resolves
+ * @param pollingInterval the time interval (in milliseconds) for how often to invoke the callback
+ * @param deps array of dependencies, which (when changed) cause the callback to be invoked again
+ * @returns an object with the result and current status of the promise
```

Comment:
I'm not sure I love making consumers use 4 variadic args like this, but it also seemed potentially nice to keep the API familiar since `useData` is common in newer components. Would it be better as an object?